packages/react-ui/src/components/organisms/AppBar/AppBar.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { AppBar as MuiAppBar, Toolbar, styled } from '@mui/material';
+
+import BurguerMenu from './BurguerMenu';
+import BrandLogo from './BrandLogo';
+import BrandText from './BrandText';
+import SecondaryText from './SecondaryText';
+
+const BrandElements = styled('div')(({ theme }) => ({
+  display: 'flex',
+  alignItems: 'center',
+  maxWidth: 'calc(100% - 300px)',
+  overflow: 'hidden',
+
+  [theme.breakpoints.down('sm')]: {
+    minWidth: '188px'
+  }
+}));
+
+const Content = styled('div')(({ theme }) => ({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'space-between',
+  flex: 1,
+  marginLeft: theme.spacing(1)
+}));
+
+const AppBar = ({
+  children,
+  brandLogo,
+  brandText,
+  secondaryText,
+  showBurgerMenu,
+  onClickMenu,
+  ...otherProps
+}) => {
+  return (
+    <MuiAppBar {...otherProps}>
+      <Toolbar>
+        <BrandElements>
+          {showBurgerMenu && <BurguerMenu onClickMenu={onClickMenu} />}
+          {brandLogo && <BrandLogo logo={brandLogo} />}
```

Comment:
I think that separating _trivial_ components like `BrandLogo`,  `BrandText` abd `SecondaryText` into sparate files/components is just too much.

They could be just `<BrandLogo>{logo}</BrandLogo>` ... where `BrandLogo` is just direct reference to local, small `styled` component without intermediate "functional" component.

Maybe it is ok in this special case, but it's not necessarily good blueprint for whole project to follow.

Reply:
@zbigg I think I want to preserve this, and have a healthy discussion about new explicit rule: one file per component element